pkg/apis/core/v1alpha1/types_cloudprofile.go

```diff
@@ -83,6 +83,16 @@ type CloudProfileSpec struct {
 	VolumeTypes []VolumeType `json:"volumeTypes,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,9,rep,name=volumeTypes"`
 }
 
+// SeedSelector contains constraints for selecting seed to be usable for shoots using a profile
+type SeedSelector struct {
+	// LabelSelector is optional and can be used to select seeds by their label settings
+	// +optional
+	*metav1.LabelSelector `json:",inline,omitempty" protobuf:"bytes,1,opt,name=labelSelector"`
```

Comment:
I cannot find an example in k/k for embedded pointer field with tags `,inline,omitempty`. The only field with tags `,inline,omitempty` is this one https://github.com/kubernetes/kubernetes/blob/d8799658082ce4e3733968042248de024bd0154e/staging/src/k8s.io/api/networking/v1beta1/types.go#L154-L160 and it is not a pointer.

The non-pointer way works:
```golang
	// LabelSelector is optional and can be used to select seeds by their label settings
	// +optional
	metav1.LabelSelector `json:",inline,omitempty" protobuf:"bytes,1,opt,name=labelSelector"`
```

It looks like the go-to-protobuf generator does not (or cannot) handle embedded field of type poitner.

Reply:
Should we then simply duplicate the content of the `LabelSelector` until this limitation is fixed?

```golang
type SeedSelector struct {
	// MatchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
  // map is equivalent to an element of matchExpressions, whose key field is "key", the
  // operator is "In", and the values array contains only "value". The requirements are ANDed.
  // +optional
  MatchLabels map[string]string `json:"matchLabels,omitempty" protobuf:"bytes,1,rep,name=matchLabels"`
  // MatchExpressions is a list of label selector requirements. The requirements are ANDed.
  // +optional
  MatchExpressions []metav1.LabelSelectorRequirement `json:"matchExpressions,omitempty" protobuf:"bytes,2,rep,name=matchExpressions"`
	// Providers is optional and can be used by restricting seeds by their provider type. '*' can be used to enable seeds regardless of their provider type.
	// +optional
	Providers []string `json:"providers,omitempty" protobuf:"bytes,3,rep,name=providers"`
}

Reply:
It is also possible to omit the `*`, This has the same problem, all the coding checks for a nil pointer (optional), so all the tests and others must be adapted.

Reply:
I finally managed to fix the protobuf generator ( see comment below)